ext/bestitamazonpay4oxid_payment.php

```diff
@@ -138,11 +142,7 @@ protected function _managePrimaryUserData($oAmazonData)
             //If our logged in user is the one that was created by us before update details from Amazon WS
             //(Can be selected another user from Amazon Address widget)
             if ($sUserAmazonOrderReferenceId === $sAmazonOrderReferenceId) {
-                $oUser->assign(array(
```

Comment:
Ich bin mir nicht sicher ob wir hier die Logik grundlegend ändern und das nicht so gedacht war das hier nur die Stadt und das Land geupdatet werden anstelle des kompletten Users. Bitte vorher mit Amazon klären.

Reply:
Laut dem Kommentar "If our logged in user is the one that was created by us before update details from Amazon WS" wird halt zu wenig geupdatet. Mit der vorherigen Anpassung kommt es dazu, dass wenn man eine komplett andere Adresse im Widget wählt, hierbei nur die neue PLZ, die neue Stadt und das neue Land übernommen wird, die Straße mit Hausnummer etc. jedoch von der alten Adresse bestehen bleibt. Das führt dann schlussendlich dazu, dass eine falsche Adresse nach Bestellung in der Shop DB landet (eher selten, da $sUserAmazonOrderReferenceId === $sAmazonOrderReferenceId zutreffen muss).

Reply:
Wie gesagt ich würde das mit Amazon klären, die haben das so abgenommen und dafür wird es ja einen Grund geben.